app/models/miq_compare.rb

```diff
@@ -223,7 +223,7 @@ def prepare_master_list
       unless @include[section].key?(:master_index)
         @include[section][:master_index] = @master_list.length
 
-        @master_list << {:name => section, :header => section_header, :group => @include[section][:group]} << (key.nil? ? nil : []) << []
+        @master_list << {:name => section, :header => section_header, :group => @include[section][:group]} << (key || []) << []
```

Comment:
This one is actually a change in behavior.  It was resulting in `nil` or `[]`, with your change, it's resulting in `key` or `[]`.  The tests are failing because of this.